src/ImageSharp/Processing/Extensions/Binarization/BinaryThresholdExtensions.cs

```diff
@@ -28,12 +29,14 @@ public static class BinaryThresholdExtensions
         /// <param name="rectangle">
         /// The <see cref="Rectangle"/> structure that specifies the portion of the image object to alter.
         /// </param>
+        /// <param name="valueType">The color component to be compared to threshold.</param>
         /// <returns>The <see cref="IImageProcessingContext"/> to allow chaining of operations.</returns>
         public static IImageProcessingContext BinaryThreshold(
             this IImageProcessingContext source,
             float threshold,
-            Rectangle rectangle) =>
-            source.ApplyProcessor(new BinaryThresholdProcessor(threshold), rectangle);
+            Rectangle rectangle,
```

Comment:
Additionally, `Rectangle` should be the last argument when there `treshold` and/or `valueType`  is present (for consistency).

Reply:
Ok, thank you for clarification. I will implement it soon.

One more question: Code coverage test continues to fail even though I have extended the test methods to pass the branch with saturation. What am I doing wrong?

Reply:
I think the problem is on the codecov (tool) side not on yours. I'm not an expert on that, @JimBobSquarePants any idea what's going on?

Reply:
What’s happening with that new test? We’re doing a new comparison with saturation but I do not see how since the submodule has not been updated.

Reply:
There was no validation in the old tests, so there is none in the new one (just debug save).

Reply:
https://github.com/SixLabors/ImageSharp/pull/1540/files#diff-88fae6a5fdda36fa3f6f2ba33c87415448b8db16c5320bdf437ee75235381be1R83

Reply:
I’m on my phone so can’t check what that verify method is doing

Reply:
If memory serves well, that method just checks if the area outside the rect was left unchanged.